t/06_events.t

```diff
@@ -97,5 +97,65 @@ subtest collect_status => sub {
   is(Mojo::IOLoop::ReadWriteProcess::Session->singleton->all->size,         1);
 };
 
+subtest collect_from_signal_handler => sub {
+  my $p = process(execute => '/usr/bin/true');
+  my $collected = 0;
+  my $orphan = 0;
+  my $sig_chld = 0;
+  $p->session->reset();
+  $p->session->collect_status(1);
+  $p->session->on(SIG_CHLD => sub { $sig_chld++});
+  $p->session->on(collected => sub { $collected++ });
+  $p->session->on(collected_orphan => sub { $orphan++ });
+  $p->start();
+
+  sleep 1;
```

Comment:
Maybe it makes more sense to sleep in a loop but with a smaller delay? So we don't always waste one seconds here but can also cope with a bit longer execution times (e.g. when the system the test runs on is very slow/busy). The same counts for the other sleeps. Additionally, it makes sense to make such timeouts configurable (e.g. via an environment variable) and only hard-code the default.

Reply:
Ok, my intention was to make the test as easy as possible. With such a loop, someone might get the feeling that it is related. 
But sure, I will just make it faster by looping over $collected. I could even just don't `sleep` at all. The disadvantage is, that such test then consume a lot of cpu in error case. 
Would be nice to have some cool stuff like `OpenQA::Test::TimeLimit` in here.

Reply:
Checking every `0.1` seconds is usually not such a big deal (CPU consumption wise) and I wouldn't extend the timeout too long (5 seconds should be enough).

When https://github.com/mudler/Mojo-IOLoop-ReadWriteProcess/issues/34 is accepted we might take the effort and improve the project a bit more, e.g. introduce `OpenQA::Test::TimeLimit` here as well.

Reply:
I will use a patched version of `Mojo::IOLoop::ReadWriteProcess::Test::Utils::attempt`